xdsl/dialects/csl/csl_stencil.py

```diff
@@ -121,6 +141,237 @@ def __init__(
+                TensorType(
+                    self.iter_arg.type.get_element_type(),
+                    (self.iter_arg.type.get_shape()[0] // self.num_chunks.value.data,),
+                ),
+                (len(self.swaps),),
+            ),
+            IndexType(),
+            self.iter_arg.type,
+        ]
+        post_process_req_types = [
+            self.communicated_stencil.type,
+            self.iter_arg.type,
+        ]
+        for arg, expected_type in zip(
+            self.chunk_reduce.block.args, chunk_reduce_req_types
+        ):
+            if arg.type != expected_type:
+                raise VerifyException(
+                    f"Unexpected block argument type of chunk_reduce, got {arg.type} != {expected_type} at index {arg.index}"
+                )
+        for arg, expected_type in zip(
+            self.post_process.block.args, post_process_req_types
+        ):
+            if arg.type != expected_type:
+                raise VerifyException(
+                    f"Unexpected block argument type of post_process, got {arg.type} != {expected_type} at index {arg.index}"
+                )
+
+        if len(self.res) < 1:
+            raise VerifyException(
+                f"Expected stencil.apply to have at least 1 result, got {len(self.res)}"
+            )
+        res_type = cast(stencil.TempType[Attribute], self.res[0].type)
+        for other in self.res[1:]:
+            other = cast(stencil.TempType[Attribute], other.type)
+            if res_type.bounds != other.bounds:
+                raise VerifyException("Expected all output types bounds to be equals.")
+
+    def get_rank(self) -> int:
+        res_type = self.res[0].type
+        assert isa(res_type, stencil.TempType[Attribute])
+        return res_type.get_num_dims()
+
+    def get_accesses(self) -> Iterable[stencil.AccessPattern]:
+        """
+        Return the access patterns of each input.
+
+         - An offset is a tuple describing a relative access
+         - An access pattern is a class wrappoing a sequence of offsets
+         - This method returns an access pattern for each stencil
+           field of the apply operation.
+        """
+        # iterate over the block arguments
+        for arg in self.chunk_reduce.block.args:
+            accesses: list[tuple[int, ...]] = []
+            # walk the uses of the argument
+            for use in arg.uses:
+                # filter out all non access ops
+                if not isinstance(use.operation, AccessOp):
+                    continue
+                access: AccessOp = use.operation
+                # grab the offsets as a tuple[int, ...]
+                offsets = tuple(access.offset)
+                # account for offset_mappings:
+                if access.offset_mapping is not None:
+                    offsets = tuple(offsets[i] for i in access.offset_mapping)
+                accesses.append(offsets)
+            yield stencil.AccessPattern(tuple(accesses))
```

Comment:
Can you add a test that this is still working as expected?

Reply:
Ok